### ISSUE TITLE
qml-ruby linker error 

### ISSUE THREAD

**Issue by shayneoneill** [oracle/truffleruby#1751]
I'm trying to use the qml-ruby library from here: https://github.com/seanchas116/ruby-qml

This works fairly reliably with regular ruby, but when I try and install (via truffleruby's bundle or gem) for for truffleruby I get this;-

```
compiling qml.c
compiling signal_emitter.c
linking shared-object qml/qml.su
/usr/lib/jvm/graalvm-ce-19.2.0.1/jre/languages/ruby/lib/cext/linker.rb:34:in `block in process_args': -rpath needs to be followed by a directory name
(RuntimeError)
	from /usr/lib/jvm/graalvm-ce-19.2.0.1/jre/languages/ruby/lib/cext/linker.rb:64:in `[]'
	from /usr/lib/jvm/graalvm-ce-19.2.0.1/jre/languages/ruby/lib/cext/linker.rb:64:in `process_args'
	from /usr/lib/jvm/graalvm-ce-19.2.0.1/jre/languages/ruby/lib/cext/linker.rb:62:in `process_args'
	from /usr/lib/jvm/graalvm-ce-19.2.0.1/jre/languages/ruby/lib/cext/linker.rb:21:in `main'
	from /usr/lib/jvm/graalvm-ce-19.2.0.1/jre/languages/ruby/lib/cext/linker.rb:150:in `<main>'
make: *** [Makefile:639: qml.su] Error 1
```

It would seem I'm supposed to provide this -rpath argument, but I must confess I'm not sure how. or if theres a deeper problem not immediately solveable by the correct combination of command line arguments, chants and incantations.

**Comment by eregon** [comment]
Thank you for the report.
I suspect this an issue in our linker and our new approach to linking which should be merged soon should resolve this.

**Comment by shayneoneill** [comment]
Spectacular. Is there an issue I can watch that would give me updates on this?

**Comment by eregon** [comment]
I'll mention it here when the approach to linking is merged, so just watching this issue should be enough.

To be clear, I don't know if qml-ruby will work after that's merged, but I think the specific error above should no longer be an issue.

**Comment by eregon** [comment]
The new linker approach has been merged and should solve this.
It will be in the next release, 19.3.0.
I cannot easily verify this as installing Qt seems rather involved.
I'll ping you when 19.3.0 is released to check whether it's fixed.

**Comment by eregon** [comment]
I'll close this issue since I think this is fixed for 19.3. I'll write here when 19.3 is released for verification.

**Comment by eregon** [comment]
@shayneoneill TruffleRuby 19.3.0 was released, could you try installing qml-ruby again?
